tmt/utils.py

```diff
@@ -368,8 +368,504 @@ def to_spec(self) -> dict[str, Any]:
+        """
+
+        root = root or Path.cwd()
+
+        result = Environment()
+
+        for filename in filenames:
+            result.update(cls.from_file(filename=filename, root=root, logger=logger))
+
+        return result
+
+    @classmethod
+    def from_inputs(
+            cls,
+            *,
+            raw_fmf_environment: Any = None,
+            raw_fmf_environment_files: Any = None,
+            raw_cli_environment: Any = None,
+            raw_cli_environment_files: Any = None,
+            file_root: Optional[Path] = None,
+            key_address: Optional[str] = None,
+            logger: tmt.log.Logger) -> 'Environment':
+        """
+        Extract environment variables from various sources.
+
+        Combines various raw sources into a set of environment variables. Calls
+        necessary functions to process environment files, dictionaries and CLI
+        inputs.
+
+        All inputs are optional, and there is a clear order of preference, which is,
+        from the most prefered:
+
+        * ``--environment`` CLI option (``raw_cli_environment``)
+        * ``--environment-file`` CLI option (``raw_cli_environment_files``)
+        * ``environment`` fmf key (``raw_fmf_environment``)
+        * ``environment-file`` fmf key (``raw_fmf_environment_files``)
+
+        :param raw_fmf_environment: content of ``environment`` fmf key. ``None``
+            and a dictionary are accepted.
+        :param raw_fmf_environment_files: content of ``environment-file`` fmf key.
+            ``None`` and a list of paths are accepted.
+        :param raw_cli_environment: content of ``--environment`` CLI option.
+            ``None``, a tuple or a list are accepted.
+        :param raw_cli_environment_files: content of `--environment-file`` CLI
+            option. ``None``, a tuple or a list are accepted.
+        :raises NormalizationError: when an input is of a type which is not allowed
+            for that particular source.
+        """
+
+        key_address_prefix = f'{key_address}:' if key_address else ''
+
+        from_fmf_files = Environment()
+        from_fmf_dict = Environment()
+        from_cli_files = Environment()
+        from_cli = Environment()
+
+        if raw_fmf_environment_files is None:
+            pass
+        elif isinstance(raw_fmf_environment_files, list):
+            from_fmf_files = cls.from_files(
+                filenames=raw_fmf_environment_files,
+                root=file_root,
+                logger=logger)
+        else:
+            raise NormalizationError(
+                f'{key_address_prefix}environment-file',
+                raw_fmf_environment_files,
+                'unset or a list of paths')
+
+        if raw_fmf_environment is None:
+            pass
+        elif isinstance(raw_fmf_environment, dict):
+            from_fmf_dict = Environment({
+                str(key): EnvvarValue(str(value))
+                for key, value in raw_fmf_environment.items()})
```

Comment:
Would it make sense to also have an `Environment.from_dict()` class method and use it here?

Reply:
Thanks!